relay-server/src/endpoints/common.rs

```diff
@@ -262,28 +263,30 @@ fn queue_envelope(
 ) -> Result<(), BadStoreRequest> {
     let envelope = managed_envelope.envelope_mut();
 
-    // Remove metrics from the envelope and queue them directly on the project's `Aggregator`.
-    let is_metric = |i: &Item| matches!(i.ty(), ItemType::Statsd | ItemType::MetricBuckets);
-    let metric_items = envelope.take_items_by(is_metric);
-
-    if !metric_items.is_empty() {
-        relay_log::trace!("sending metrics into processing queue");
-        state.processor().send(ProcessMetrics {
-            items: metric_items.into_vec(),
-            project_key: envelope.meta().public_key(),
-            start_time: envelope.meta().start_time(),
-            sent_at: envelope.sent_at(),
-        });
-    }
+    if state.config().relay_mode() != RelayMode::Proxy {
+        // Remove metrics from the envelope and queue them directly on the project's `Aggregator`.
```

Comment:
```suggestion
        // Remove metrics from the envelope and queue them directly on the project's `Aggregator`. In proxy mode, we cannot aggregate metrics because we may not have a project ID.
```